src/marionette.rs

```diff
@@ -334,11 +334,12 @@ impl MarionetteHandler {
                          .map_err(|_| WebDriverError::new(ErrorStatus::UnknownError,
                                                           "Unable to read profile preferences file")));
 
-        prefs.insert("marionette.defaultPrefs.port", Pref::new(port as i64));
+        for &(ref name, ref value) in prefs::DEFAULT.iter() {
```

Comment:
My thinking at the moment is to move those preferences that are absolutely required to meet WebDriver conformance from `prefs::DEFAULT` to `prefs::REQUIRED` and to accept this change. This would change the meaning of ‘default preferences’ towards ‘suggested preferences’, as they would not take effect if a custom profile with custom preferences is provided.